independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/project/buildfile/AbstractGradleBuildFile.java

```diff
@@ -36,81 +34,79 @@ public AbstractGradleBuildFile(final Path projectDirPath, final QuarkusPlatformD
                         out.toByteArray());
             }
         } else {
-            writeToProjectFile(SETTINGS_GRADLE_PATH, getModel().getSettingsContent().getBytes());
+            writeToProjectFile(getSettingsGradlePath(), getModel().getSettingsContent().getBytes());
             try (ByteArrayOutputStream out = new ByteArrayOutputStream()) {
                 getModel().getPropertiesContent().store(out, "Gradle properties");
                 writeToProjectFile(GRADLE_PROPERTIES_PATH, out.toByteArray());
             }
         }
-        writeToProjectFile(BUILD_GRADLE_PATH, getModel().getBuildContent().getBytes());
-    }
-
-    @Override
-    protected boolean addDependency(AppArtifactCoords coords, boolean managed) {
-        return addDependencyInModel(getModel(), coords, managed);
+        writeToProjectFile(getBuildGradlePath(), getModel().getBuildContent().getBytes());
     }
 
-    static boolean addDependencyInModel(Model model, AppArtifactCoords coords, boolean managed) {
+    static String createDependencyCoordinatesString(Model model, AppArtifactCoords coords, boolean managed) {
         boolean isBOM = "pom".equals(coords.getType());
-        StringBuilder newDependency;
-        if (isBOM) {
-            // Check if BOM is not included already
-            String resolvedPlatform = String
-                    .format("%s:%s", getProperty(model, "quarkusPlatformGroupId"),
-                            getProperty(model, "quarkusPlatformArtifactId"));
-            String thisBOM = String.format("%s:%s", coords.getGroupId(), coords.getArtifactId());
-            if (thisBOM.equals(resolvedPlatform)) {
-                // BOM matches the platform, no need to do anything
-                return false;
-            }
-            newDependency = new StringBuilder()
-                    .append("    implementation enforcedPlatform(\"")
-                    .append(thisBOM).append(":").append(coords.getVersion())
-                    .append("\")'");
-        } else {
-            newDependency = new StringBuilder()
-                    .append("    implementation '")
-                    .append(coords.getGroupId())
-                    .append(":")
-                    .append(coords.getArtifactId());
-        }
+        // Special case for platform dependency. We will create properties based dependency string.
+        // It will be ignored if it is already in "dependencies" section.
+        // Note that we ignore version here - if someone tries to add platform with different version, we will still use
+        // version from properties.
+        if (isBOM && !managed
+                && Objects.equals(coords.getGroupId(), getProperty(model, "quarkusPlatformGroupId"))
+                && Objects.equals(coords.getArtifactId(), getProperty(model, "quarkusPlatformArtifactId"))) {
+            return "enforcedPlatform(\"${quarkusPlatformGroupId}:${quarkusPlatformArtifactId}:${quarkusPlatformVersion}\")";
```

Comment:
This seem to be only `Kotlin` compatible no?

Reply:
This part should be fine for Groovy and Kotlin DSL (property names are same), but this still will not work since AbstractGroovyGradleBuildFile and AbstractKotlinGradleBuildFile will surround this string with ' and ". Not sure why integration tests didn't catch that. I will fix this.